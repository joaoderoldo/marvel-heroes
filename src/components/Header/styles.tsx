import styled from "styled-components";

export const HeaderWrapper = styled.header`
  background-color: var(--white-color, #fff);
  height: 65px;
  padding: 16px 0;

  .header-container {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  .user {
    display: flex;
    align-items: center;

    .user-content {
      text-align: right;
      margin-right: 8px;

      > .name {
        font-weight: bold;
        font-size: 0.875rem;
        line-height: 0.875rem;
      }

      > .subtitle {
        font-size: 0.625rem;
        line-height: 0.625rem;
        margin-top: 8px;
      }

      > .name,
      > .subtitle {
        display: block;
      }
    }
  }

  .user-avatar {
    width: 42px;
    height: 42px;
    border-radius: 9999px;
    background-color: var(--primary, black);
    color: var(--white, #fff);
    font-size: 1.25rem;
    display: flex;
    justify-content: center;
    align-items: center;
    font-weight: bold;
  }
`;
