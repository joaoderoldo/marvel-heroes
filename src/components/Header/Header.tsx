import { Link } from "react-router-dom";

import { ReactComponent as Logo } from "@/assets/svgs/neural-med-logo.svg";
import { Container } from "@/components";
import { getInitialLetters } from "@/utils";

import { HeaderWrapper } from "./styles";

const Header = () => {
  return (
    <HeaderWrapper test-id="header-component">
      <Container classes="header-container">
        <Link to="/">
          <Logo test-id="logo-header-component" />
        </Link>
        <div className="user">
          <div className="user-content">
            <span className="name">João Vitor</span>
            <div className="subtitle">Teste de Front-end</div>
          </div>
          <div className="user-avatar">{getInitialLetters("João Vitor")}</div>
        </div>
      </Container>
    </HeaderWrapper>
  );
};

export default Header;
