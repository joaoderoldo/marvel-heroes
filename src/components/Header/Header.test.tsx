import { render, waitFor } from "@testing-library/react";
import Header from "@/components/Header/Header";

import { BrowserRouter } from "react-router-dom";

describe("Header component", () => {
  it("renders correctly and match snapshot", () => {
    const { queryByTestId } = render(<Header />, {
      wrapper: BrowserRouter,
    });
    expect(queryByTestId("header-component")).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByTestId } = render(<Header />, {
      wrapper: BrowserRouter,
    });
    waitFor(() => expect(getByTestId("header-component")).toBeInTheDocument());
  });

  it("renders the logo", () => {
    const { getByTestId } = render(<Header />, { wrapper: BrowserRouter });
    waitFor(() =>
      expect(getByTestId("logo-header-component")).toBeInTheDocument()
    );
  });
});
