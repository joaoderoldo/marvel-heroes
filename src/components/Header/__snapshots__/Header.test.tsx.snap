// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders correctly and match snapshot 1`] = `null`;
